### ISSUE TITLE
Some accessibility bugs

### ISSUE THREAD

**Issue by RafaelFernandesBR** [laurent22/joplin#8794]
Hello, I'd like to report some accessibility bugs I've noticed with the Joplin app for Android.

Firstly, when navigating the app with a screen reader, visually, the note titles are displayed for the user to click and open the note to be read. However, with screen readers like [TalkBack](https://play.google.com/store/apps/details?id=com.google.android.marvin.talkback) or Jieshu, we cannot simply tap anywhere on the screen and land on a note title. We have to swipe the entire screen using right/left navigation gestures until we find the note we want to open.

The correct behavior should be that the notes are displayed across the entire screen, and if the user clicks anywhere on the screen where a note is located, the title should be read aloud.

Another bug is that the app has several unlabeled buttons. For example, when opening a note, there's a button to access options like sharing, setting a reminder (in the case of a task), etc. This button is not read by the screen reader; it's as if there's nothing there. The user only knows it's a button if they are curious enough to click and see what it does.

The correct approach would be to label the button to indicate its purpose.

When clicking the "Add New" button in the lower-right corner of the screen, one of the few buttons we can access directly without swiping the screen, it displays buttons for adding a note or a task. For screen readers, it shows 3 buttons for adding a note and 3 for adding a task.

The correct approach would be to have only one button for each action.

## Environment

Joplin version: 2.12.2
Platform: Android
OS specifics: Android 13

## Steps to reproduce

1. Open Joplin.
2. Navigate through the notes using a screen reader, using direct touch on the screen.
3. Notice that when navigating, no notes are displayed; you can only navigate using right/left gestures of the screen reader.
4. With a note open, you'll notice that the action button lacks a label for the screen reader.
5. When clicking the button to add something new, verify that for the screen reader, it displays 6 buttons, 3 for a new note, and 3 for a new task.

**Comment by personalizedrefrigerator** [comment]
I've created a [separate issue](https://github.com/laurent22/joplin/issues/8796) for the missing screen reader labels issue raised above (that way this issue can remain open after missing labels have been added).

**Comment by personalizedrefrigerator** [comment]
Duplicate new note/to-do buttons and broken direct touch selection seem to be related to [the react-native-paper library](https://reactnativepaper.com/) (we use `react-native-paper` to create the new note/to-do buttons). If `react-native-paper` is removed, direct touch works.

I'm in the process of creating an issue in the [react-native-paper GitHub repository](https://github.com/callstack/react-native-paper/issues).
**Edit**: Upstream bug reports: https://github.com/callstack/react-native-paper/issues/4064, https://github.com/callstack/react-native-paper/issues/4063

I think direct touch used to work with `react-native-paper` — perhaps we can revert to a previous version of the library until the issue is fixed.

**Comment by personalizedrefrigerator** [comment]
Still an issue.

**Comment by RafaelFernandesBR** [comment]
Not yet resolved

Em qua., 8 de nov. de 2023 13:02, github-actions[bot] <
***@***.***> escreveu:

**Comment by personalizedrefrigerator** [comment]
Only partially resolved.

Work to be done:
- [ ] Fix the "new note" button (will likely require upstream contributions to `react-native-paper`)
- [ ] Fix the "sort by" modal (focus doesn't jump to the modal). This may also require upstream contributions (or switching libraries).
- [ ] Fix focus not moving to the "note actions" menu.

**Comment by personalizedrefrigerator** [comment]
https://github.com/callstack/react-native-paper/issues/4063 was recently closed upstream. Upgrading react-native-paper should thus address that issue in Joplin.

**Comment by RafaelFernandesBR** [comment]
It hasn't been fixed yet

**Comment by personalizedrefrigerator** [comment]
Should be partially fixed by: https://github.com/laurent22/joplin/pull/10123

**Comment by RafaelFernandesBR** [comment]
Problem fixed in version 3.0.1
